### ISSUE TITLE
rework concept of post-fix mappings to components and subcomponents

### ISSUE THREAD

**Issue by Ryex** [Ryex/pyitect#14]
post-fix mappings are kinda dodgy.

there needs to be a better to differentiate different providers of the same component

introducing component tags!

currently components can have any name you want

from here on however dot notation will indicate sub-component types

`someFuncProvider` is a generic component
`someFuncProvider.variant1` is STILL `someFuncProvider` generic component. but it's of the `variant1` variety

these types can be stacked up
`A.B.C.E.F.G` is still valid
and `A.B.C.E.F.G` is also an `A.B.C.E.F` and a `A.B.C.E` and even an `A`

basically sub-classes but without multiplei inheritance
